### PR TITLE
Feature/pelagos 3503 reject requeue item if dataservers are down or 500

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: src/Pelagos/Bundle/AppBundle/Rabbit/Consumer/DoiConsumer.php [hunk 1]
```diff
@@ -68,18 +71,20 @@ public function __construct(
...
      *
-     * @return boolean True if success, false otherwise.
+     * @return integer
      */
     public function execute(AMQPMessage $message)
     {
         // @codingStandardsIgnoreStart
         $routingKey = $message->delivery_info['routing_key'];
 
+        $msgStatus = self::MSG_ACK;
```
Comment (mickel1138) [inline src/Pelagos/Bundle/AppBundle/Rabbit/Consumer/DoiConsumer.php]: Aren't these states ConsumerInterface::MSG_ACK ?

See: https://github.com/php-amqplib/RabbitMqBundle/blob/master/RabbitMq/ConsumerInterface.php

Comment (praneethpr) [reply src/Pelagos/Bundle/AppBundle/Rabbit/Consumer/DoiConsumer.php]: Yeah, these states are from ConsumerInterface and DoiConsumer implements this interface

File: src/Pelagos/Util/DOIutil.php [hunk 2]
```diff
@@ -112,9 +116,14 @@ public function mintDOI(
...
 
         //check to see if it worked.
-        //using in array because EZID API returns 201 and EZDatacite API returns 200.
+        //using in array because EZID API returns 201 and EZ Datacite API returns 200.
         if (!in_array($httpCode, [200, 201])) {
-            throw new \Exception("ezid failed with:$httpCode($output)", $httpCode);
+            $expMsg = "ezid failed with:$httpCode($output)";
+            if ($httpCode >= 400) {
+                throw new HttpClientErrorException($expMsg, $httpCode);
+            } elseif ($httpCode >= 500) {
```
Comment (fightingtexasaggie) [inline src/Pelagos/Util/DOIutil.php]: This condition can never be reached.

File: src/Pelagos/Util/DOIutil.php [hunk 3]
```diff
@@ -181,7 +191,12 @@ public function createDOI(
         //check to see if it worked.
         //using in array because EZID API returns 201 and EZDatacite API returns 200.
         if (!in_array($httpCode, [200, 201])) {
-            throw new \Exception("ezid failed with:$httpCode($output)", $httpCode);
+            $expMsg = "ezid failed with:$httpCode($output)";
+            if ($httpCode >= 400) {
+                throw new HttpClientErrorException($expMsg, $httpCode);
+            } elseif ($httpCode >= 500) {
```
Comment (fightingtexasaggie) [inline src/Pelagos/Util/DOIutil.php]: See above, (unreachable condition).

File: src/Pelagos/Util/DOIutil.php [hunk 4]
```diff
@@ -240,7 +256,12 @@ public function updateDOI(
 
         //check to see if it worked.
         if (200 != $httpCode) {
-            throw new \Exception("ezid failed with:$httpCode($output)", $httpCode);
+            $expMsg = "ezid failed with:$httpCode($output)";
+            if ($httpCode >= 400) {
+                throw new HttpClientErrorException($expMsg, $httpCode);
+            } elseif ($httpCode >= 500) {
```
Comment (fightingtexasaggie) [inline src/Pelagos/Util/DOIutil.php]: here too

File: src/Pelagos/Util/DOIutil.php [hunk 5]
```diff
@@ -268,7 +290,12 @@ public function getDOIMetadata($doi)
...
```
Comment (fightingtexasaggie) [inline src/Pelagos/Util/DOIutil.php]: and here

File: src/Pelagos/Util/DOIutil.php [hunk 6]
```diff
@@ -308,7 +336,12 @@ public function deleteDOI($doi)
...
```
Comment (fightingtexasaggie) [inline src/Pelagos/Util/DOIutil.php]: one last one

**Comment by praneethpr** [comment]
> Can we make the sleep time a class constant?

Sure, I didn't think about it. Thanks Mike W

**Pull request merged**